src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -729,11 +755,33 @@ void AppendEncoding (string encoding, ref List <string> list)
 			return httpConnection;
 		}
 
+		/// <summary>
+		/// Configure and return a custom <see cref="t:SSLSocketFactory"/> for the passed HTTPS <paramref
+		/// name="connection"/>. If the class overriding the method returns anything but the default
+		/// <c>null</c>, the SSL setup code will not call the <see cref="ConfigureKeyManagerFactory"/> nor the
+		/// <see cref="ConfigureTrustManagerFactory"/> methods used to configure a custom trust manager which is
+		/// then used to create a default socket factory.
+		/// Deriving class must perform all the key manager and trust manager configuration to ensure proper
+		/// operation of the returned socket factory.
+		/// </summary>
+		/// <returns>Instance of SSLSocketFactory ready to use with the HTTPS connection.</returns>
+		/// <param name="connection">HTTPS connection to return socket factory for</param>
+		protected virtual SSLSocketFactory ConfigureCustomSSLSocketFactory (HttpsURLConnection connection)
```

Comment:
Should this instead be called `CreateCustomSslSocketFactory()`? `Configure`, to me, implies that it's configuring an *existing* instance/state, not creating and returning *new* state. (Based on return type, I'm assuming it can't be configuring existing state.)